### PR TITLE
Surface underlying exception msg at plugin load

### PR DESCRIPTION
**Pull request by leondz**
we should get the underlying exception into the log.

Example -- traces like this are unhelpful:

```
2024-08-15 17:14:15,770  INFO  probe queue: probes.topic.WordnetControversial
2024-08-15 17:14:15,770  INFO  probe init: <garak.probes.topic.WordnetControversial object at 0x744e06e45880>
2024-08-15 17:14:15,770  WARNING  Exception failed instantiation of garak.probes.topic.WordnetControversial
2024-08-15 17:14:15,770  WARNING  failed to load probe ValueError('Plugin WordnetControversial not found in probes.topic'):
2024-08-15 17:14:15,770  INFO  run complete, ending
2024-08-15 17:14:15,788  INFO  garak run complete in 3.48s
```

& there is a typically a more helpful message available:

```
(garak) 17:14:35 x1:~/dev/garak [feature/topic_probe] $ python
Python 3.12.4 | packaged by Anaconda, Inc. | (main, Jun 18 2024, 15:12:24) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import garak
>>> import garak._config
>>> garak._plugins
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'garak' has no attribute '_plugins'
>>> import garak._plugins
>>> garak._plugins.lo
garak._plugins.load_plugin(  garak._plugins.logging       
>>> garak._plugins.lo
garak._plugins.load_plugin(  garak._plugins.logging       
>>> garak._plugins.load_plugin("probes.topic.WordnetControversial")
WARNING:root:Exception failed instantiation of garak.probes.topic.WordnetControversial
Traceback (most recent call last):
  File "/home/lderczynski/dev/garak/garak/_plugins.py", line 394, in load_plugin
    plugin_instance = klass(config_root=config_root)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lderczynski/dev/garak/garak/probes/topic.py", line 93, in __init__
    wn.config.downloads_directory = _config.transient.cache_dir / "wn" / "downloads"
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: property 'downloads_directory' of 'WNConfig' object has no setter

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/lderczynski/dev/garak/garak/_plugins.py", line 400, in load_plugin
    raise ValueError(
ValueError: Plugin WordnetControversial not found in probes.topic
>>> 
```
this PR places the underlying message, in this example `AttributeError: property 'downloads_directory' of 'WNConfig' object has no setter`, into `garak.log`.

### CODE REVIEW

**Comment by leondz** [comment]
resolves #40

**Review by kbmlcoding — APPROVED** [review]

**Review by jmartin-tech — APPROVED** [review]
I like it, one optional minor comment about possibly using `logging.exception` to allow the exception context to log `e`. The change would increase the message level from `warning` to `error`.

File: garak/_plugins.py [hunk 1]
```diff
@@ -388,24 +388,13 @@ def load_plugin(path, break_on_fail=True, config_root=_config) -> object:
...
-        )
-        if break_on_fail:
-            raise ValueError(
-                f"Plugin {plugin_class_name} not found in {category}.{module_name}"
-            ) from ae
-        else:
-            return False
     except Exception as e:
-        # print("error in: module", mod.__name__, "class", plugin_class_name)
         logging.warning(
```
[the remark this answers is not in the document]
Comment (leondz) [reply garak/_plugins.py:396]: I think the docs might be slightly misleading here: `logging` messages use the same interface as `debug()`, which prints trace if given `exc_info`. But only `exception` gets an extra bit of text in its method summary saying that it does this.

tl;dr: can do this without even changing the level ☺️
